pkg/controller/worker/machine_crds.go

```diff
@@ -297,20 +298,36 @@ func init() {
 	}
 }
 
+// EnsureMachineResourcesForConfig ensures that all well-known machine CRDs are created or updated.
+func EnsureMachineResourcesForConfig(ctx context.Context, config *rest.Config) error {
```

Comment:
How about having this in k8s style and calling it `ApplyMachineResourcesWithConfig`? Same goes for the function `EnsureMachineResources` -> `ApplyMachineResources`.